nasa/harmony
# HARMONY-1573: Test harmony in a box in github actions

## Jira Issue ID
HARMONY-1573

## Description
This PR uses colima to run harmony in a box in github actions and verify a basic harmony request works. When running locally, colima can replace docker as the container runtime to run harmony in a box and harmony dev system as well.


## Local Test Steps
To test running harmony in a  box using colima as container runtime locally:
1) shut down Docker Desktop
2) brew install colima
3) colima start --with-kubernetes
4) Make sure `LOCAL_DEV=false` in .env, run `KUBE_CONTEXT=colima bin/bootstrap-harmony`
5) verify all pods running successfully in k8s with `kubectl -n harmony get pods` 
6) Verify harmony request works with e.g. `http://localhost:3000/C1233800302-EEDTEST/ogc-api-coverages/1.0.0/collections/all/coverage/rangeset?granuleId=G1233800343-EEDTEST&format=image%2Fpng` and `http://localhost:3000/workflow-ui`

Also verify running harmony locally still works with Docker Desktop as container runtime by shutting down colima and start Docker Desktop and harmony the same way as before.

Currently, the workflow action takes over 2 hours to run and is sometimes flaky when running in github, we may want to configure this workflow as `workflow_dispatch` to run it manually on demand. 
## PR Acceptance Checklist
